src/util/range.h

```diff
@@ -353,6 +353,16 @@ struct ranget final
     }
   }
 
+  /// Return an new range containing the same elements except for the first
+  /// \p count elements.
+  /// If the range has fewer elements, returns an empty range.
+  ranget<iteratort> skip(std::size_t count) const
```

Comment:
I don't find the names `drop` and `skip` entirely intuitive. I am wondering whether using longer, more unambiguous names would be better?

Reply:
Maybe. I'm open to suggestions. `drop` is also used by range-v3, so we should maybe keep this one.

Reply:
Whatever is used elsewhere I'm happy with, just stick with that terminology then.

Reply:
`drop` and `skip` are both fine names, but they are kinda synonyms. I think we should choose one or the other, but not both. The name used in `.net`'s linq is `skip` - https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.skip?view=netframework-4.7.2
The name in haskell would be drop - http://zvon.org/other/haskell/Outputprelude/drop_f.html
Because C++ supports overloading we could actually have 3 different overloads, all with the same name if we wanted. A mutation overload, a `const &` overload and an r-value overload. I would suggest keeping just a `const &` and the r-value overload, in order to reduce the need to reason about mutation.